acme/client.go

```diff
@@ -408,9 +408,11 @@ func (c *Client) solveChallenges(challenges []authorizationResource) map[string]
 			for i, solver := range solvers {
 				// TODO: do not immediately fail if one domain fails to validate.
 				err := solver.Solve(authz.Body.Challenges[i], authz.Domain)
-				if err != nil {
-					failures[authz.Domain] = err
+				if err == nil {
+					// we're done
+					return nil
```

Comment:
This function finds all appropriate solvers for all given domains and executes them. Each entry in the challenges array stands for one domain. Returning early after the first successful solve will definitely break multiple domains.